### DISCUSSION TITLE
Lateral join

### DISCUSSION THREAD

**Discussion by dullbananas** [diesel-rs/diesel#3858] [Q&A]
How can I do a `LATERAL JOIN` without using custom SQL for anything other than the `FROM` clause?

**Comment by weiznich** [comment]
`JOIN LATERAL` is currently not supported by the build in DSL. That means you need to write a custom query DSL extension for that. See [the extending diesel guide](https://diesel.rs/guides/extending-diesel.html) for details.

**Comment by dullbananas** [comment]
It might be possible to make a custom FROM clause by putting it in a custom expression that's used as argument for the diesel::select function

**Comment by ChStark** [comment]
As an extra not completely related comment, it could be amazing if diesel could achieve something similar to what zapatos does with joins and nested json (https://jawj.github.io/zapatos/#joins-as-nested-json)  it creates a single query to fetch all the object tree (in most of the cases increasing performance being the network a bottleneck)
